### DISCUSSION TITLE
reusing SecondQuantizedMolecule

### DISCUSSION THREAD

**Discussion by jmaguiar** [goodchemistryco/Tangelo#374] [Q&A] [answered]
is it possible to reuse a previous SecondQuantizedMolecule computation? I conduct some experiments in a jupyter notebook. It takes some minutes to have the result of SecondQuantizedMolecule and I wonder if storing the results and reading in later experiments woudl be faster

**Comment by alexfleury-sb** [comment, marked as answer]
As of now, this is not possible. 

The naive way would be to use the `pickle` python module to do so, but it is not compatible with `SecondQuantizedMolecule` (related to https://github.com/goodchemistryco/Tangelo/issues/361).

An interface we thought of was to be able to read `FCIDUMP` file, which [can be outputted by Psi4](https://psicode.org/psi4manual/master/api/psi4.driver.fcidump.html), PySCF, ORCA, Gaussian, and many more. The workflow would be to perform a SCF calculation, save the results to an FCIDUMP file, and then create quantum inputs with the provided file in Tangelo.

**Comment by ValentinS4t1qbit** [comment]
I would definitely avoid pickle, because it breaks as soon as version numbers in environment etc changes. I like the idea of using a default format supported by the quantum chemistry tools, which empowers users to use a lot of tools outside of Tangelo in their overall workflow. 

Edit: Would the FCIDUMP feature be complicated to implement, and how useful do we think it is ?

**Reply by alexfleury-sb** [reply]
I don't think it would be complicated to implement, as we could tap into [`pyscf` helper functions](https://github.com/pyscf/pyscf/blob/d3f622d46eef5d9b8702fa6f4577babfb6c2ccfe/pyscf/tools/fcidump.py#L242). I tested the function with Psi4 a while ago, and the PySCF function needed a small fix (we should probably push it to the PySCF repo). It may be more difficult to test other packages for us  as they often require commercial licenses, but as is it would support the saving of SCF calculation in files for PySCF + Psi4.

The bulk of the work would be to code a new `IntegralSolver` that instantiates the right objects, and is compatible with our solver. Maybe it would also require to do a cleaning job in `SecondQuantizedMolecule`.

**Reply by ValentinS4t1qbit** [reply]
I wouldn't worry about other commercial packages now, and simply focus on PySCF and Psi4.

As to how to proceed, I naively thought that adding methods to `IntegralSolver` to enable it to export a`FCIDUMP` file or have a new init method `from_fcidump` or something was going to be enough, but we could have a quick chat for me to better understand that.
